src/bp/cluster.ts

```diff
@@ -30,10 +36,19 @@ export const setupMasterNode = (logger: sdk.Logger) => {
     worker.kill()
   })
 
-  cluster.on('exit', (worker, code, signal) => {
+  cluster.on('exit', async (worker, code, signal) => {
+    const workerIdx = process.ML_WORKERS.indexOf(worker.id)
+    if (workerIdx !== -1) {
+      debug(`Machine learning worker ${worker.id} died`)
+      process.ML_WORKERS.splice(workerIdx, 1)
+      if (process.ML_WORKERS.length === 0) {
+        await spawnMLWorkers(logger)
```

Comment:
Shouldn't we want to spawn one as soon as it dies ? Here you would slowly lose your nodes, until only one processes all the training